### ISSUE TITLE
Allow test pane to use any cwl-runner

### ISSUE THREAD

**Issue by kaushik-work** [rabix/composer#420]
# Requirements
1. User should be able to set path to `cwl-runner` in RC configuration file
1. If this is set, when user clicks RUN button on test pane, RC will invoke given `cwl-runner`
1. RC will supply cwl-runner name, --outdir path, CWL doc and JOB YAML.
1. This will be added to the docs.

# Specifications
If the user places the following key in the configuration file
```
{
    "runner": "my-cwl-runner",
}
```

RC will then do an OS call as

```
my-cwl-runner --outdir whatever-rc-sets my.cwl job.yaml
```

# Not doing for this implementation
1. No need for integration with RC (throbber, logs etc). Just need to be able to run task, set output directory and then allow user to see output dir. Logs button inactive
1. No need for UI for changing config file (but great if possible)

**Comment by kaushik-work** [comment]
@portah I recall you had an effort related to this?

**Comment by michael-kotliar** [comment]
Hey, @kaushik-work, I'm interested in this feature too and I have a related question.
Does Rabix Composer run executor for every workflow step separately or executor is called only once for the whole workflow?

**Comment by michael-kotliar** [comment]
As a temporary solution, `cwltool` can be run as custom executor when it's wrapped in Java. Here is the jar file [custom-executor.jar](https://github.com/michael-kotliar/custom-executor/releases/download/0.0.2/custom-executor.jar) that runs `cwltool` using `--outdir` `app` and `job` provided by Rabix Composer. It also supports `--enable-composer-logs` to highlight current progress.

**Comment by kaushik-work** [comment]
@michael-kotliar thank you so much for this contribution. I've opened a few issues as feedback after experimenting with it. I'm not a Java coder, but I can try and help as I can. 

This is very exciting, because I can see this working, and then what I'll do is update the documentation on Desktop Rabix Composer telling people where to download the custom executors from and how to configure DRC/the executor to get up and running.

Thanks again!

**Comment by kaushik-work** [comment]
This is implemented here: https://github.com/gungorbudak/composer/tree/custom-local-executor

And will be merged during the next release after appropriate QA.

**Comment by ionox0** [comment]
🎊

**Comment by kaushik-work** [comment]
Closed by 2cbedfaaf43b4a88ee97ed5c896717376c090ab3
